scheduler/scheduler_label_test.go

```diff
@@ -84,4 +90,8 @@ func (s *SchedulerTestSuite) TestLabelFilter(c *check.C) {
 	actual, err = scheduler.PrioritizeCandidates(nil, context7)
 	c.Assert(err, check.IsNil)
 	c.Assert(actual, check.DeepEquals, []string{})
+
+	actual, err = scheduler.PrioritizeCandidates(nil, context8)
+	c.Assert(err, check.IsNil)
+	c.Assert(actual, check.DeepEquals, []string{})
```

Comment:
@StrongMonkey Your test doesn't quite represent the scenarios we need to cover.

Here's what we want to prove works:

When host A has label: `require_any: "foo=bar,foo1=bar1,foo5"`, if context has label `foo5=blah`, host A *is valid*

Also, just thinking about the multiple instance context, if any instance in the context has the foo5 label key then the host should be valid. We don't need to see the label on every instance in the context. Please add a test for multiple instances in the context.